TESTING.md

```diff
@@ -0,0 +1,159 @@
+
+### Running the tests
+
+Once you have finished building and "adding" your box (with `vagrant box add`),
+you'll need to modify the `.kitchen.yml`. The only modifications you should
+need to make are replacing our box names with yours. For example, you would
+replace `apex/macos-10.13.3` with `my_high_sierra_box`. To double check the
+available boxes and their names, execute `vagrant box list`. For example:
+
+```shell
+$ vagrant box list
+apex/macos-10.11.6 (parallels, 1.0.0)
+apex/macos-10.12.6 (parallels, 2.0.0)
+apex/macos-10.13.3 (parallels, 1.1.0)
+```
+
+Next, make sure you're in the macOS cookbook root and run `kitchen list` to view
+the available instances. It should look something like this:
+
+```shell
+$ kitchen list
+Instance                      Driver   Provisioner  Verifier  Transport  Last Action    Last Error
+default-apex-macos-10133      Vagrant  ChefZero     Inspec    Ssh        <Not Created>  <None>
+default-apex-macos-10126      Vagrant  ChefZero     Inspec    Ssh        <Not Created>  <None>
+default-apex-macos-10116      Vagrant  ChefZero     Inspec    Ssh        <Not Created>  <None>
+xcode-apex-macos-10133        Vagrant  ChefZero     Inspec    Ssh        <Not Created>  <None>
+xcode-apex-macos-10126        Vagrant  ChefZero     Inspec    Ssh        <Not Created>  <None>
+xcode-apex-macos-10116        Vagrant  ChefZero     Inspec    Ssh        <Not Created>  <None>
+spotlight-apex-macos-10133    Vagrant  ChefZero     Inspec    Ssh        <Not Created>  <None>
+spotlight-apex-macos-10126    Vagrant  ChefZero     Inspec    Ssh        <Not Created>  <None>
+spotlight-apex-macos-10116    Vagrant  ChefZero     Inspec    Ssh        <Not Created>  <None>
+certificate-apex-macos-10133  Vagrant  ChefZero     Inspec    Ssh        <Not Created>  <None>
+certificate-apex-macos-10126  Vagrant  ChefZero     Inspec    Ssh        <Not Created>  <None>
+certificate-apex-macos-10116  Vagrant  ChefZero     Inspec    Ssh        <Not Created>  <None>
+```
+
+The `kitchen list` command serves as a nearly-perfect way to validate the
+`.kitchen.yml` syntax. For more info, check out `kitchen help` for commands and
+run `kitchen help COMMAND` for help on a specific subcommand. When you're ready,
+run `kitchen test`.
+
+```shell
+$ kitchen test
+```
+
+`kitchen` supports using regular expressions to only run a specific instance.
+For example:
+
+```shell
+$ kitchen test 1011 # test all the suites on 10.11 only
+$ kitchen test xcode # test the xcode suite on all versions
+$ kitchen test default.*101[23] # only test default suites on 10.12 and 10.13
+```
+
+macOS takes a little while to boot and the suites themselves (especially Xcode)
+can take a while to run - some of our builds end up being 30-40 minutes per operating
+system. If you've got the hardware, don't be afraid to run
+`kitchen test --concurrency n` to save a little time (where `n` is the number of concurrent
+instances you want to boot up).
+
+## Testing via a Pull Request
```

Comment:
I don't feel that it's a good idea to offer up our infrastructure as a path to testing their changes. My concern is that we have other jobs that run on our Mac build machines, and if they're tied up running tests for someone else's change it could have an impact.

Reply:
Let’s see what the actual utilization looks like before we decide this.

Reply:
I can see both points. Right now, we don't have a lot of PRs being opened up - but that is presumably only going to increase. Ultimately, I think this section is better suited for a `CONTRIBUTING.md` though, so I'm going to remove from `TESTING.md` until that is implemented.